src/sdk/BotFrameworkConfig.ts

```diff
@@ -56,16 +56,87 @@ export class BotFrameworkConfig extends DialogServiceConfigImpl {
         botFrameworkConfig.setProperty(PropertyId.SpeechServiceConnection_Region, region);
+
+        if (botId) {
+            botFrameworkConfig.setProperty(PropertyId.Conversation_ApplicationId, botId);
+        }
+
+        return botFrameworkConfig;
+    }
+
+    /**
+     * Creates an instance of a BotFrameworkConfig.
+     * This method is intended only for users who use a non-default service host. The standard resource path will be assumed.
+     * For services with a non-standard resource path or no path at all, use fromEndpoint instead.
+     * Note: Query parameters are not allowed in the host URI and must be set by other APIs.
+     * Note: To use an authorization token with fromHost, use fromHost(URL),
+     * and then set the AuthorizationToken property on the created BotFrameworkConfig instance.
+     * Note: Added in version 1.9.0.
+     * @member BotFrameworkConfig.fromHost
+     * @function
+     * @public
+     * @param {URL | string} host - If a URL is provided, the fully-qualified host with protocol (e.g. wss://your.host.com:1234)
+     *  will be used. If a string is provided, it will be embedded in wss://{host}.convai.speech.azure.us.
+     * @param {string} subscriptionKey - The subscription key. If a subscription key is not specified, an authorization token must be set.
+     * @param botId Optional. Identifier for using a specific bot within an Azure resource group. Equivalent to the resource name.
+     * @returns {BotFrameworkConfig} A speech factory instance.
+     */
+    public static fromHost(
+        host: URL | string,
+        subscriptionKey?: string,
+        botId?: string): BotFrameworkConfig {
+
+        Contracts.throwIfNullOrUndefined(host, "host");
+        const resolvedHost: URL = host instanceof URL ? host : new URL(`wss://${host}.convai.speech.azure.us`);
+        Contracts.throwIfNullOrUndefined(resolvedHost, "resolvedHost");
+
+        const botFrameworkConfig: BotFrameworkConfig = new DialogServiceConfigImpl();
+        botFrameworkConfig.setProperty(PropertyId.Conversation_DialogType, DialogServiceConfig.DialogTypes.BotFramework);
+        botFrameworkConfig.setProperty(PropertyId.SpeechServiceConnection_Host, resolvedHost.toString());
+
+        if (undefined !== subscriptionKey) {
+            botFrameworkConfig.setProperty(PropertyId.SpeechServiceConnection_Key, subscriptionKey);
+        }
+
+        return botFrameworkConfig;
+    }
+
+    /**
+     * Creates an instance of a BotFrameworkConfig.
+     * This method is intended only for users who use a non-standard service endpoint or parameters.
+     * Note: The query parameters specified in the endpoint URL are not changed, even if they are set by any other APIs.
+     * Note: To use authorization token with fromEndpoint, pass an empty string to the subscriptionKey in the
+     *       fromEndpoint method, and then set authorizationToken="token" on the created BotFrameworkConfig instance to
+     *       use the authorization token.
```

Comment:
Probably should add version note here as well

Reply:
Agreed, will for sure do that.